### PR TITLE
feat: graphql query for recommended search questions

### PR DESCRIPTION
**Pull request by sshanzel**
A new query to fetch recommended search questions based on the user's 5 recent upvotes then randomized the result to fetch only 3 items.

WT-1647 #done

### CODE REVIEW

File: src/schema/posts.ts [hunk 1]
```diff
@@ -1002,6 +1013,35 @@ export const resolvers: IResolvers<any, Context> = {
         },
       );
     },
+    searchQuestionRecommendations: async (
+      source,
+      _,
+      ctx: Context,
+    ): Promise<GQLPostQuestion[]> => {
+      const res = await ctx.con
```
Comment (rebelchris) [inline src/schema/posts.ts:1021]: Would graphorm not be preferred here?

Comment (sshanzel) [reply src/schema/posts.ts:1021]: Makes sense. Let me see how it would be able to easily work with the complex we have here.

Comment (sshanzel) [reply src/schema/posts.ts]: Done! 🥳

**Review by rebelchris — APPROVED** [review]
Looks good to me 🙏

File: src/schema/posts.ts [hunk 2]
```diff
@@ -469,6 +472,12 @@ export const typeDefs = /* GraphQL */ `
     image: String!
   }
 
+  type PostQuestion {
+    id: String!
+    postId: String!
```
Comment (idoshamun) [inline src/schema/posts.ts:477]: This needs to be a post object

File: src/schema/posts.ts [hunk 3]
```diff
@@ -1002,6 +1013,31 @@ export const resolvers: IResolvers<any, Context> = {
...
+                .from(Upvote, 'u')
+                .where({ userId: ctx.userId })
+                .orderBy('u."createdAt"', 'DESC')
+                .limit(5),
+            'upvoted',
+            `"${builder.alias}"."postId" = upvoted."postId"`,
+          )
+          .orderBy('rand', 'DESC')
+          .limit(3)
+          .getRawMany(),
```
Comment (idoshamun) [inline src/schema/posts.ts:1038]: This is not needed for graphorm

File: src/schema/posts.ts [hunk 4]
```diff
@@ -1002,6 +1013,31 @@ export const resolvers: IResolvers<any, Context> = {
...
     },
+    searchQuestionRecommendations: async (
+      source,
+      _,
+      ctx: Context,
+      info,
+    ): Promise<GQLPostQuestion[]> =>
+      graphorm.query(ctx, info, (builder) => ({
+        queryBuilder: builder.queryBuilder
+          .addSelect('random()', 'rand')
```
Comment (idoshamun) [inline src/schema/posts.ts:1024]: We don't need to add it as a select. We can put it directly in the order by

Comment (sshanzel) [reply src/schema/posts.ts]: Oh, cool. Nice to know we can do that.

**Review by idoshamun — APPROVED** [review]
Amazing!

**Pull request merged**